Added greater agent capabilities, upload media (image/video) + quote tweets (with or without media) + following users, re-tweeting, & liking posts (for the algo !)

kingbootoshi opened the pull request:
# Enhanced Twitter Client Features

This PR adds several new features to the Twitter client to make capabilities more autonomous

## New Features
- Media uploads (photos/videos) support
- Quote tweets functionality
- Following users capability
- Tweet liking functionality

## Technical Details
- Added multi-step video upload process with chunked uploading
- Implemented media processing status checks
- Added rate limit handling for follow operations

## Testing
All new features have been tested, included a command.ts file (optional) which allows user's to test functionalities via a CLI. Added node-fetch as an NPM package to test uploading images/videos from a link.  

## Notes
- Maintains compatibility with existing features
- No breaking changes to the current API

lastchancedegen commented:
based

lalalune commented:
i fucking love this dev

kingbootoshi commented:
> i fucking love this dev

based, just wanted to do my part. inspired by your work and movement!

lalalune commented on src/tweets.ts:
```diff
@@ -163,100 +163,45 @@ export async function fetchTweetsAndReplies(
+  auth: TwitterAuth
+): Promise<Response> {
+  const features = {
+    // ... existing features ...
```
clauded

lalalune commented:
okay reviewed

- noticed one spot where we might have lost come code due to clauding
- some merge conflicts in tweets

kingbootoshi commented:
Cool, pulled the latest repo changes and updated the code. Since someone handled media upload in the latest pull, I built the new functions I added on top of that. 

there should be no claude or missing code, now things are only added in their proper position (QT/Likes/RTs in tweet.ts, follows in relationships.ts)

Check it out now ! Optimized and cleaned up, I also added testing for the new functions that passed.

vbhakta8 commented on README.md:
```diff
@@ -1,11 +1,11 @@
-# agent-twitter-client
+# goat-x
```
Probably want to change this back to existing
```suggestion
# agent-twitter-client
```

vbhakta8 commented on README.md:
````diff
@@ -1,11 +1,11 @@
 
 ```sh
-npm install agent-twitter-client
+npm install goat-x
````
As well as here
```suggestion
npm install agent-twitter-client
```

kingbootoshi commented:
oops! yea those changes weren't supposed to go on this branch. thanks !
